keps/sig-auth/2579-psp-replacement/README.md

```diff
@@ -0,0 +1,1086 @@
+role must be unset. Spec:
+
+- SELinuxOptions.Type
+    - **Restricted Fields:**
+        - `spec.securityContext.seLinuxOptions.type`
+        - `spec.containers[*].securityContext.seLinuxOptions.type`
+        - `spec.initContainers[*].securityContext.seLinuxOptions.type`
+    - **Allowed Values:**
+        - undefined/empty
+        - `container_t`
+        - `container_init_t`
+        - `container_kvm_t`
+
+- SELinuxOptions.User and SELinuxOptions.Role
+    - **Restricted Fields:**
+        - `spec.securityContext.seLinuxOptions.user`
+        - `spec.containers[*].securityContext.seLinuxOptions.user`
+        - `spec.initContainers[*].securityContext.seLinuxOptions.user`
+        - `spec.securityContext.seLinuxOptions.role`
+        - `spec.containers[*].securityContext.seLinuxOptions.role`
+        - `spec.initContainers[*].securityContext.seLinuxOptions.role`
+    - **Allowed Values:** undefined/empty
+
+- SELinuxOptions.Level
+    - Unrestricted.
+
+**Non-root Groups** - (restricted) This optional constraint will be omitted from the initial
+implementation.
+
+**Seccomp**
+- (restricted) Must be set to anything except `unconfined`. Same reasoning as AppArmor. If the default
+  value changes from `unconfined` (see https://github.com/kubernetes/enhancements/issues/2413), then
+  the requirement to set a profile will be lifted.
+- (baseline) May be unset; any profile except `unconfined` allowed.
+
+<<[UNRESOLVED]>>
+
+**BLOCKING**
+
+**Capabilities** - (baseline) Only the following capabilities may be added:
+- AUDIT_WRITE
+- CHOWN
+- DAC_OVERRIDE
+- FOWNER
+- FSETID
+- KILL
+- MKNOD
+- NET_BIND_SERVICE
+- SETFCAP
+- SETGID
+- SETPCAP
+- SETUID
+- SYS_CHROOT
+
+Notes:
+- This set is equal to the [docker default capability
+  set](https://docs.docker.com/engine/reference/run/#runtime-privilege-and-linux-capabilities) minus
+  `NET_RAW`.
+- The [OpenShift default capability
+  set](https://github.com/openshift/hypershift-toolkit/blob/148be6f31a365dbcdf1cbd647773f59ccbcc282a/assets/ignition/files/etc/crio/crio.conf#L87-L102)
+  also drops AUDIT_WRITE, MKNOD, SETFCAP, and SYS_CHROOT._
+- The allowed set for the restricted profile is implicitly empty (no capabilities), since
+  [Kubernetes does not support ambient
+  capabilities](https://github.com/kubernetes/kubernetes/issues/56374). If ambient capability
+  support is ever added, we may want to consider a more conservative allowed set for the restricted
+  profile.
+
+---
+
+Prior discussion:
+
+Kubernetes does not define the default set, but Docker does. This leaves us with 2 options:
+
+1. Hardcode the [docker default capability
+   set](https://docs.docker.com/engine/reference/run/#runtime-privilege-and-linux-capabilities), and
+   allow adding a capability in that set. This enables workloads to use an allowlist approach where
+   they drop all capabilities (`ALL`), and only explicitly add the ones they need. However, it means
```

Comment:
Just a side note: dropping all capabilities is not portable, since ALL is not recognized as capability name by all runtimes https://bugzilla.redhat.com/show_bug.cgi?id=1931838

Reply:
Oh interesting. Good to know.